src/zarr/v3/store/local.py

```diff
@@ -67,6 +67,9 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"LocalStore({repr(str(self))})"
 
+    def __eq__(self, other: 'LocalStore') -> bool:
```

Comment:
After adding `from typing import Self` to the top of the file, I think you can do this:

```suggestion
    def __eq__(self, other: Self) -> bool:
```